front_end/src/backend/downloads.js

```diff
@@ -14,98 +14,70 @@
+            if (!(fileName in FILESIZE_BYTES) || !(fileName in FILESIZE_MIB)) {
+                throw new Error("Invalid File Name");
+            }
+            if (!(bucket in REGIONS_MAP)) {
+                throw new Error("Invalid Bucket Name");
+            }
+        }
 
-async function benchmarkSingleDownload(fileName, bucketName) {
-    let result = new Map();
-    const timeTaken = await timeDownload(fileName, bucketName);
+        const bucketName = 'gcsrbpa-' + bucket;
 
-    let fileSizeBytes = 0;
-    let fileSizeMiB = 0;
-    if (fileName == '2mib.txt') {
-        fileSizeBytes = 2097152;
-        fileSizeMiB = 2;
-    } else if (fileName == '64mib.txt') {
-        fileSizeBytes = 67108864;
-        fileSizeMiB = 64;
-    } else if (fileName == '256mib.txt') {
-        fileSizeBytes = 268435456;
-        fileSizeMiB = 256;
+        const URL = `https://storage.googleapis.com/${bucketName}/${fileName}`;
+
+        const timeTaken = await this.downloadFile(URL);
+        return timeTaken / 1000; // return in units of seconds
     }
-    const speedBps = fileSizeBytes / timeTaken;
-    const speedMiBps = fileSizeMiB / timeTaken;
 
-    result.set('bucketName', bucketName);
-    result.set('location', REGIONS_MAP[bucketName]);
-    result.set('fileName', fileName);
-    result.set('timeTaken', timeTaken);
-    result.set('fileSizeBytes', fileSizeBytes.toString());
-    result.set('speedBps', speedBps.toFixed(3));
-    result.set('speedMiBps', speedMiBps.toFixed(3));
+    async benchmarkSingleDownload(fileName, bucketName) {
+        let result = new Map();
+        const timeTaken = await this.timeDownload(fileName, bucketName);
 
-    return result;
-}
+        let fileSizeBytes = FILESIZE_BYTES[fileName] || fileName;
+        let fileSizeMiB = FILESIZE_MIB[fileName] || fileName;
+        let location = REGIONS_MAP[bucketName] || bucketName;
 
-export async function benchmarkDownload(fileName, bucketName) {
-    const result = await benchmarkSingleDownload(fileName, bucketName, REGIONS_MAP);
+        const speedBps = (fileSizeBytes / timeTaken) || -1;
+        const speedMiBps = (fileSizeMiB / timeTaken) || -1;
 
-    console.log(`Completed Downloads Benchmark for ${bucketName}`);
+        result.set('bucketName', bucketName);
+        result.set('location', location);
+        result.set('fileName', fileName);
+        result.set('timeTaken', timeTaken);
+        result.set('fileSizeBytes', fileSizeBytes);
+        result.set('speedBps', speedBps.toFixed(3));
+        result.set('speedMiBps', speedMiBps.toFixed(3));
 
-    let arr = new Array()
-    arr.push(Object.fromEntries(result))
-    return JSON.stringify(arr);
-}
+        return result;
+    }
+
+    async benchmarkDownload(fileName, bucketName) {
+        const result = await this.benchmarkSingleDownload(fileName, bucketName);
+
+        let arr = new Array();
+        arr.push(Object.fromEntries(result));
+        return JSON.stringify(arr);
```

Comment:
I may have asked this previously but is there a reason to pass stringified representations around as opposed to just returning the array?

Reply:
I'm not sure why, but returning an object directly seemed to result in `undefined` being returned. Returning a string that gets parsed later fixed it.

Reply:
I'll pull the code next week and take a look.